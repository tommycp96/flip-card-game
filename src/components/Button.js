import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import { styles } from '../styles/common';

const Button = ({ title, onPress }) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text style={{ fontSize: 14, fontWeight: 'bold' }}>{title}</Text>
    </TouchableOpacity>
  );
};

export default Button;
