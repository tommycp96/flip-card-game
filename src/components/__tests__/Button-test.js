import 'react-native';
import React from 'react';
import Button from '../Button';
import renderer from 'react-test-renderer';

it('should renders correctly', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
