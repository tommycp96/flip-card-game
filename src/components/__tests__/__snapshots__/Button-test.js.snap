// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4fc3f7",
      "borderRadius": 8,
      "elevation": 4,
      "justifyContent": "center",
      "marginVertical": 8,
      "opacity": 1,
      "padding": 8,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.22,
      "shadowRadius": 2.22,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  />
</View>
`;
